### ISSUE TITLE
Snap package support (Feature/Enhancement)

### ISSUE THREAD

**Issue by Mikerr1111** [gradle/gradle#7430]
I was thinking the setup and install process for Ubuntu and other Linux distros could be improved by having a [Snap package](https://snapcraft.io/) version. If it's not a priority would you mind if I implemented the snap?

[Creating a Snap](https://tutorials.ubuntu.com/tutorial/create-your-first-snap#0)

[Create snaps from your preferred programming language.](https://docs.snapcraft.io/build-snaps/languages)

[Linux distros with Snap support](https://docs.snapcraft.io/core/install)
![snap](https://user-images.githubusercontent.com/18353476/43989596-ec564800-9d01-11e8-89e6-8e0af1434f62.png)

**Comment by big-guy** [comment]
@Mikerr1111 do you mean a Gradle distribution snap package or a Gradle plugin to produce Snaps or something else?

**Comment by Mikerr1111** [comment]
a Gradle distribution snap package @big-guy

**Comment by Mikerr1111** [comment]
Here's how the Build and Test process works for snaps. https://snapcraft.io/build

**Comment by eriwen** [comment]
Thanks for the info @Mikerr1111.

This seems useful, but I can't quite tell if this is popular enough to be needed. I feel like Gradle needs to own the distribution to ensure consumer confidence.

Perhaps there are others who would vouch for this technology?

**Comment by Mikerr1111** [comment]
Hi @eriwen, some companies that made snaps to distribute their application's across multiple Linux distributions include JetBrains, Microsoft, Amazon, Mozilla, and Spotify.  https://snapcraft.io/
![snap](https://user-images.githubusercontent.com/18353476/47349441-7c96ca80-d668-11e8-8bd0-4bcfd36db5d2.png)

**Comment by eriwen** [comment]
Yes, I see. This seems valuable and I think we should do this eventually, but I don't yet see enough folks that would get that value. 

What I really want to know is: Why should we work on this instead of the other 1100 issues we have open?

**Comment by lacasseio** [comment]
@eriwen We had other requests in the past like this and it's hard to keep them up as well as introducing new ones. What I thought is to have a separate repository just for delivery. This way when someone wants to have a new delivery, it could be easy to create a PR with the plumbing. As for the maintenance, given the repo would be small-ish the community could maintain this central repo and we can just jump in to help with the direction and modeling. We would own the account the packaging system and the rest could be taken care of by the community. What do you think?

@Mikerr1111 Would that be something you would want to help with?

**Comment by Mikerr1111** [comment]
@lacasseio, I'll help in any way I can.
